dune/grid/polyhedralgrid/grid.hh

```diff
@@ -0,0 +1,1207 @@
+    {
+      const int index = seed.index();
+      switch (codim)
+      {
+        case 0:
+          return 1;
+        case 1:
+          return grid_.cell_facepos[ index+1 ] - grid_.cell_facepos[ index ];
+        case dim:
+          return cellVertices_[ index ].size();
+      }
+      return 0;
+    }
+
+    template <int codim>
+    typename Codim<codim>::EntitySeed
+    subEntitySeed( const typename Codim<0>::EntitySeed& elemSeed, const int i ) const
+    {
+      assert( i>= 0 && i<subEntities( elemSeed, codim ) );
+      typedef typename Codim<codim>::EntitySeed  EntitySeed;
+      if( codim == 0 )
+        return EntitySeed( elemSeed.index() );
+      else if ( codim == 1 )
+      {
+        return EntitySeed( grid_.cell_faces[ grid_.cell_facepos[ elemSeed.index() ] + i ] );
+      }
+      else if ( codim == dim )
+      {
+        return EntitySeed( cellVertices_[ elemSeed.index() ][ i ] );
+      }
+      else
+      {
+        DUNE_THROW(InvalidStateException,"codimension not availalbe");
+      }
+    }
+
+    const std::vector< GeometryType > &geomTypes ( int codim ) const
+    {
+      static std::vector< GeometryType > emptyDummy;
+      if (0 <= codim && codim < geomTypes_.size())
+        return geomTypes_[codim];
+
+      return emptyDummy;
+    }
+
+    int indexInInside( const typename Codim<0>::EntitySeed& seed, const int i ) const
+    {
+      if( ! grid_.cell_facetag )
+        return i;
+      else
+      {
+        // assert( i>= 0 && i<subEntities( EntitySeed( seed.index() ) ) );
+        return grid_.cell_facetag[ grid_.cell_facepos[ seed.index() ] + i ] ;
+      }
+    }
+
+    typename Codim<0>::EntitySeed
+    neighbor( const typename Codim<0>::EntitySeed& seed, const int i ) const
+    {
+      typedef typename Codim<0>::EntitySeed EntitySeed;
+      const int face = 2 * this->template subEntitySeed<1>( seed, i ).index();
+      int nb = grid_.face_cells[ face ];
+      if( nb == seed.index() )
+        nb = grid_.face_cells[ face+1 ];
+
+      return EntitySeed( nb );
+    }
+
+    int
+    indexInOutside( const typename Codim<0>::EntitySeed& seed, const int i ) const
+    {
+      typedef typename Codim<0>::EntitySeed EntitySeed;
+      EntitySeed nb = neighbor( seed, i );
+      const int faces = subEntities( seed, 1 );
+      for( int face = 0; face<faces; ++ face )
+      {
+        if( neighbor( nb, face ).equals(seed) )
+          return indexInInside( nb, face );
+      }
+      DUNE_THROW(InvalidStateException,"inverse intersection not found");
+      return -1;
+    }
+
+    template <class EntitySeed>
+    GlobalCoordinate
+    outerNormal( const EntitySeed& seed, const int i ) const
+    {
+      const int face  = this->template subEntitySeed<1>( seed, i ).index();
+      const int normalIdx = face * GlobalCoordinate :: dimension ;
+      GlobalCoordinate normal = copyToGlobalCoordinate( grid_.face_normals + normalIdx );
+      const int nb = grid_.face_cells[ face ];
```

Comment:
Here you need to use `2*face`. Also, braces for the `if`. (I will now stop nagging about this, and assume you will fix all remaining un-nagged-about cases...)